server/bleep/src/agent/symbol.rs

```diff
@@ -0,0 +1,370 @@
+                        "symbol": {
+                            "type": "integer",
+                            "description": "The chunk alias"
+                        }
+                    },
+                    "required": ["symbol"]
+                }
+            }])
+        )
+        .unwrap();
+
+        let llm_response = match self.llm_with_function_call(prompt, filter_function).await {
+            Ok(llm_response) => llm_response,
+            Err(e) => {
+                warn!(
+                    "Symbol classifier llm call failed, picking the first symbol: {}",
+                    e
+                );
+                llm_gateway::api::FunctionCall {
+                    name: Some("filter".to_string()),
+                    arguments: "{\"symbol\": 0}".to_string(),
+                }
+            }
+        };
+
+        let filter_argument: Filter =
+            match serde_json::from_str(llm_response.clone().arguments.as_str()) {
+                Ok(argument) => argument,
+                Err(_e) => {
+                    warn!("Cannot deserialize: {:?}", llm_response);
+                    return Err(SymbolError::DeserializeFilter);
+                }
+            };
+
+        let selected_symbol = filter_argument.symbol;
+
+        // finding symbol metadata
+        match symbols
+            .into_iter()
+            .flat_map(|(_, symbol_with_alias)| symbol_with_alias)
+            .find(|(alias, _)| *alias == selected_symbol as i32)
+        {
+            Some((_alias, symbol_metadata)) => Ok(symbol_metadata),
+            _ => Err(SymbolError::OutOfBounds),
+        }
+    }
+
+    pub async fn get_related_chunks(&mut self, chunks: Vec<CodeChunk>) -> Vec<CodeChunk> {
+        const MAX_CHUNKS: usize = 3;
+
+        // get symbols with ref/defs for each chunk
+        let chunks_with_symbols = futures::future::join_all(
+            chunks
+                .iter()
+                .filter(|c| !c.is_empty())
+                .map(|c| self.extract_symbols(c.clone())), // TODO: Log failure
+        )
+        .await
+        .into_iter()
+        .filter_map(Result::ok)
+        .collect();
+
+        // get original user query
+        let user_query = self.last_exchange().query.target().unwrap();
+
+        // select one symbol
+        let selected_symbol = match self.filter_symbols(&user_query, chunks_with_symbols).await {
+            Ok(selected_symbol) => selected_symbol,
+            Err(e) => {
+                info!("Returning no extra chunks: {}", e);
+                return Vec::new();
+            }
+        };
+
+        // get expanded chunks for selected symbol
+        let extra_chunks = self.expand_symbol_into_chunks(selected_symbol).await;
+
+        // take 3 chunks, update path aliases, update enchange chunks
+        let extra_chunks = extra_chunks
+            .iter()
+            .take(MAX_CHUNKS)
+            .map(|c| {
+                let chunk = CodeChunk {
```

Comment:
if we update the path alias in the `expand_symbols_into_chunks` function; we will have to simply clone the `chunk` here. if can cannot update the path alias earlier, we can use this shorthand notation:

```rust
let chunk = CodeChunk {
    alias: self.get_path_alias(..),
    ..c       // the rest of the values are cloned from `c`
}

Reply:
thanks, much cleaner now